src/Pages/AddNewProject/AddNewProject.tsx

```diff
@@ -0,0 +1,87 @@
+import { AuthContext } from '@/Context/AuthContext';
+import { ToastContext } from '@/Context/ToastContext';
+import baseUrl from '@/utils/Custom/Custom';
+import { useContext, useState } from 'react';
+import { useForm } from 'react-hook-form';
+import { MdKeyboardArrowLeft } from "react-icons/md";
+import { Link, useNavigate } from 'react-router-dom';
+import './AddNewProject.module.scss';
+import { INewProject } from '@/Interfaces';
+
+
+
+const AddNewProject = () => {
+
+  const required = "This Field is required"
+  const { handleSubmit, register, formState: { errors } } = useForm<INewProject>()
+  const [Loading, setLoading] = useState(false)
+  const navigate = useNavigate()
+
+  const toast = useContext(ToastContext)
+  const data = useContext(AuthContext)
+  let headers = data?.header
+  //!  Add New Projects
+  const handleAdd = (data: INewProject) => {
+    setLoading(true)
+    return baseUrl.post(`/api/v1/Project`, data ,{
+      headers,
+    })
+      .then(() => {
+        if (toast)
+          toast.getToastValue("success", "Project created successfully")
+        setLoading(false)
```

Comment:
Consider using finally(() => setIsLoading(false)) after the promise to consistently reset the loading state to false, avoiding redundancy in both the then and catch blocks.